src/prefect/infrastructure/kubernetes.py

```diff
@@ -378,11 +379,21 @@ def _get_cluster_uid(self) -> str:
         There is no real unique identifier for a cluster. However, the `kube-system`
         namespace is immutable and has a persistence UID that we use instead.
 
+        PREFECT_KUBERNETES_CLUSTER_UID can be set in cases where the `kube-system`
+        namespace cannot be read e.g. when a cluster role cannot be created.
+
         See https://github.com/kubernetes/kubernetes/issues/44954
         """
+        # Default to an environment variablle
```

Comment:
```suggestion
        # Default to an environment variable
```
^Minor typo, but that's secondary.

The docstring reads like we give preference to `kube-system` namespace and then go to `PREFECT_KUBERNETES_CLUSTER_UID` if that's a no go. It looks like we're giving preference to `PREFECT_KUBERNETES_CLUSTER_UID` in the case where both are provided.

Reply:
👍 addressed